Octokit.Tests/GitHubClientTests.cs

```diff
@@ -1,19 +1,41 @@
 ﻿using System;
-using System.Net.Http.Headers;
 using System.Threading.Tasks;
 using NSubstitute;
 using Octokit.Internal;
 using Xunit;
-using Xunit.Extensions;
 using System.Collections.Generic;
 
 namespace Octokit.Tests
 {
     public class GitHubClientTests
     {
-        public class TheConstructor
+        public class TheCtor
         {
             [Fact]
+            public void EnsuresNonNullArguments()
+            {
+                Assert.Throws<ArgumentNullException>(() => new GitHubClient((IConnection)null));
+                Assert.Throws<ArgumentNullException>(() => new GitHubClient((ProductHeaderValue)null));
+
+                Assert.Throws<ArgumentNullException>(() => new GitHubClient(new ProductHeaderValue("UnitTest"), (ICredentialStore)null));
```

Comment:
Do you think we could simplify these rather than creating new values everywhere they're used?

e.g.

```csharp
var header = new ProductHeaderValue("UnitTest");

Assert.Throws<ArgumentNullException>(() => new GitHubClient(header, (ICredentialStore)null));
```

This might help with the readability of these tests...